src/scan.js

```diff
@@ -0,0 +1,18 @@
+const { githubClient } = require("./lib/github");
+const { enUs } = require("./utils/words-typos");
+
+async function scanGithubRepository(repositoryUrl = "") {
+  const client = githubClient();
+  client.validateUrl(repositoryUrl);
+  await client.init();
+
+  for (const word of enUs) {
+    await client.scan(repositoryUrl, word);
+  }
+
+  await client.close();
+}
```

Comment:
Maybe it makes sense to send the url to the init() method, which in turn stores the url string and does the validations.

client: https://i.imgur.com/2CCsYnn.png

```js
async function scanGithubRepository(repositoryUrl = "") {
  const client = githubClient();

  await client.init(repositoryUrl);

  for (const word of enUs) {
    await client.scan(word);
  }

  await client.close();
}
```

Reply:
Nice. I just maintained the method signature you applied there. But ok, nice suggestion.

Reply:
There is missing the `await` keyword on the loop but the linter pointed out that this is too heavyweight, so I'm changing this to use another approach...

Reply:
Done.